### PR TITLE
Adding parallel connections

### PR DESCRIPTION
**Pull request by anandmohan777**

### CODE REVIEW

File: src/main/java/eu/elixir/ega/ebi/egafuse/EgaFuse.java [hunk 1]
```diff
@@ -705,4 +711,9 @@ public int open(String path, FuseFileInfo fi) {
 
         return 0;
     }
+
+    public static int getConnection() {
+        return connection;
+    }
```
Comment (gariem) [inline src/main/java/eu/elixir/ega/ebi/egafuse/EgaFuse.java]: Maybe we can add Lombok library so we don't need to manually write getters and setters?

File: src/main/java/eu/elixir/ega/ebi/egafuse/EgaFuse.java [hunk 2]
```diff
@@ -248,6 +251,9 @@ private static void parse(String[] args) {
             if (cmd.hasOption("p")) {       // PASSWORD specified
                 password = new GuardedString(cmd.getOptionValue("p").toCharArray());
             }
+            if (cmd.hasOption("c")) {     
+                connection = Integer.valueOf(cmd.getOptionValue("c"));
```
[the remark this answers is not in the document]
Comment (jorizci) [reply src/main/java/eu/elixir/ega/ebi/egafuse/EgaFuse.java]: Agree, but the or else shouldbe 1. At least we need 1 connections :)

But In general I would opt to rewrite the whole argument parsing and move to jopts as it has better support for scenarios where it is either A or B or C and at the same time if A is present then D is also mandatory.

File: src/main/java/eu/elixir/ega/ebi/egafuse/EgaFuse.java (hunk 1 above)
Comment (jorizci) [reply src/main/java/eu/elixir/ega/ebi/egafuse/EgaFuse.java]: As a general rule we don't use Lombok. Getters can be autogenerated on a case by case basis and we prefer to maintain control of what do we want to be accesible.

File: src/main/java/eu/elixir/ega/ebi/egafuse/filesystems/internal/EgaRemoteFile.java [hunk 3]
```diff
@@ -44,35 +43,27 @@
...
 
     public EgaRemoteFile(String name, EgaApiDirectory parent, EgaFileDto theFile) {
         super(name, parent);
         this.theFile = theFile;
         setType();
+        setCache();
+    }
 
-        // Init cache
+    private void setCache() {
```
Comment (jorizci) [inline src/main/java/eu/elixir/ega/ebi/egafuse/filesystems/internal/EgaRemoteFile.java]: This method is not a setter, it is a whole initialization, please change name accordingly

File: src/main/java/eu/elixir/ega/ebi/egafuse/EgaFuse.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (anandmohan777) [reply src/main/java/eu/elixir/ega/ebi/egafuse/EgaFuse.java]: I will do the change

File: src/main/java/eu/elixir/ega/ebi/egafuse/filesystems/internal/EgaRemoteFile.java (hunk 3 above)
Comment (anandmohan777) [reply src/main/java/eu/elixir/ega/ebi/egafuse/filesystems/internal/EgaRemoteFile.java]: > Can you clarify me one thing, is that cache per file?

Yes cache is made per file

Comment (anandmohan777) [reply src/main/java/eu/elixir/ega/ebi/egafuse/filesystems/internal/EgaRemoteFile.java]: > This method is not a setter, it is a whole initialization, please change name accordingly

Okay I will change the name

Comment (jorizci) [reply src/main/java/eu/elixir/ega/ebi/egafuse/filesystems/internal/EgaRemoteFile.java]: > > Can you clarify me one thing, is that cache per file?
> 
> Yes cache is made per file

Then I'm sorry but this approach will not work. The way it is implemented you will have N connections at most for a specific file but that means that if they read multiple files we are gonna have potentially too many connections per instance going against our system without any bound. We will need to decide in an architecture approach and redo how it has been done.

**Pull request closed without merging**